src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -208,7 +208,9 @@ private List<OptionSettingItemSummary> ListOptionSettingSummary(IOptionSettingHa
                     Value = optionSettingHandler.GetOptionSettingValue(recommendation, setting),
                     Advanced = setting.AdvancedSetting,
                     ReadOnly = recommendation.IsExistingCloudApplication && !setting.Updatable,
-                    Visible = optionSettingHandler.IsOptionSettingDisplayable(recommendation, setting),
+                    Visible =
+                        optionSettingHandler.IsOptionSettingDisplayable(recommendation, setting) &&
+                        !(recommendation.IsExistingCloudApplication && !setting.Updatable && !setting.VisibleOnRedeployment),
```

Comment:
- can test coverage be applied to this new logic 
- "visible on redeployment" sounds definitive, but this isn't the case. It appears that "Updatable" takes priority (which makes sense). How can future maintainers avoid assumptions when they set VisibleOn to true and expect their field to always appear?

Reply:
I added a server mode test to mimic the behavior in the toolkit. This change only applies to server mode.
The JSON schema for the recipe explains this. I did not want to make the setting name very long.

Reply:
Initially my changes were in `IsOptionSettingDisplayable` but that impacts both the CLI, server mode, summary view and detailed view of option settings which is not what I want. So I moved it to server-mode specifically.

Reply:
Ah, that'd be good context to add as a comment, but not a deal breaker if you're not already revisiting this file again.

Reply:
Added a comment